Week 1 663380247-9 นาย อเสข ปัญญาวงค์

Lab 1.1

In [1]:
print("Hello World")

Hello World


Lab 1.2

Declare and print variables:

In [2]:
Age = 21
GPA = 2.88
Favorite_Color = "Black"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  21 <class 'int'>
GPA =  2.88 <class 'float'>
Favorite_Color =  Black <class 'str'>
Student =  True <class 'bool'>


Perform simple arithmetic:

In [5]:
num1 = 5
num2 = 10
print(f"sum : {num1} + {num2} = {num1 + num2}")

sum : 5 + 10 = 15


String concatenation and f-strings:

In [6]:
First_name = "Asek"
Last_name = "Panyawong"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Asek Panyawong


User input and type conversion:

In [9]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
print(f"{num} + 5 = {int(num)+5}")

What your favorite number : 5
type of A is :  <class 'str'>
5 + 5 = 10


Activity 1.1: Class Discussion - Uses of Scripting

Format: Open discussion, possibly in small groups first, then full class.

Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."

Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.

Answer: "Automated daily task, create workflow conect muti task, ai deverlop/deployment"

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output

Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected).
For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [16]:
#World problems:Find triangle's area
base = input("Enter the base of the triangle: ")
height = input("Enter the height of the triangle: ")
triangle_area = 0.5 * int(base) * int(height)
print(triangle_area)
print(type(triangle_area))

Enter the base of the triangle: 3
Enter the height of the triangle: 4
6.0
<class 'float'>


In [10]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์:13.5

ทำไมถึงเป็นtype 'float' เพราะ เมื่อนำ int มากระทำทางคณิตกับ float จะได้ float

In [11]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.

ทำไมถึงเป็นtype 'str' เพราะเมื่อ int, float  มากระทำ operator กับ string จะถูกเปลี่ยนเป็น string  

In [12]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False

ทำไมถึงเป็นtype 'bool' เพราะ bool กระทำ กับ bool ด้วยกันยังไงก็ได้ bool

In [13]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: nothing

ทำไมถึงเป็นtype '' เพราะ puzzle นี่ไม่ใช่ code ที่ execute ได้ เป็นแค่ comment ใน file เท่านั้น